s3torchconnector/tst/unit/lightning/test_s3_lightning_checkpoint.py

```diff
@@ -0,0 +1,146 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+
+from io import BytesIO
+from operator import eq
+from pathlib import Path
+from typing import Callable, Any
+
+import pytest
+import torch
+from hypothesis import given
+
+from s3torchconnector._s3client import MockS3Client
+from s3torchconnector.lightning import S3LightningCheckpoint
+from .._checkpoint_utils import (
+    python_primitives,
+    byteorders,
+    save_with_byteorder,
+    load_with_byteorder,
+    _patch_byteorder,
+)
+from s3torchconnectorclient import S3Exception
+
+TEST_BUCKET = "test-bucket"
+TEST_KEY = "test-key"
+TEST_REGION = "us-east-1"
+
+
+@given(python_primitives, byteorders)
+def test_lightning_checkpointing_saves_python_primitives(data, byteorder):
+    _test_save(data, byteorder)
+
+
+@given(byteorders)
+def test_lightning_checkpointing_saves_tensor(byteorder):
+    tensor = torch.tensor([[0.1, 1.2], [2.2, 3.1], [4.9, 5.2]])
```

Comment:
Can we please use `torch.rand(...)` here and for the other occurences? 
https://pytorch.org/docs/stable/generated/torch.rand.html